odoo/http.py

```diff
@@ -415,6 +368,133 @@ def serialize_exception(exception):
         'context': getattr(exception, 'context', {}),
     }
 
+
+# =========================================================
+# File Streaming
+# =========================================================
+
+KEEP = object()  # sentinel
+
+@dataclass
+class StreamData:
+    type = ''  # 'data' or 'path' or 'url'
```

Comment:
and that's why dataclasses are terrible :(

Reply:
Could be `type: typing.Literal['data', 'path', 'url']` but as we don't use typing it was kinda useless